scertecd/scertecd.go

```diff
@@ -1004,24 +1004,33 @@ func (s *Server) makeRecord(ctx context.Context, logf logf, recordName, txtVal s
 	// but we also don't get new cert often. But we could cache them and then
 	// only reload on miss.
 	var hostedZoneID string
-	err := svc.ListHostedZonesPagesWithContext(ctx, &route53.ListHostedZonesInput{
-		MaxItems: aws.String("100"),
-	}, func(page *route53.ListHostedZonesOutput, lastPage bool) bool {
-		for _, hz := range page.HostedZones {
-			if hz.Name == nil || *hz.Name == "local." {
-				continue
-			}
-			if strings.HasSuffix(recordName, "."+strings.TrimSuffix(*hz.Name, ".")) {
-				hostedZoneID = path.Base(*hz.Id) // map "/hostedzone/ZFOO" to "ZFOO"
-				logf("matched hosted zone %q (%s)", *hz.Name, hostedZoneID)
-				return false // stop
+	for {
+		var nextMarker *string
```

Comment:
You'll need this to go outside the `for`, otherwise it's a fresh binding for each iteration of the loop.

Reply:
great spot. thanks 🫠